subsystems/networking/bluetooth.go

```diff
@@ -0,0 +1,166 @@
+package networking
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"runtime"
+	"sync"
+	"time"
+
+	"go.viam.com/rdk/logging"
+	"go.viam.com/utils"
+)
+
+// bluetoothService provides methods to retrieve cloud config and/or WiFi credentials for a robot over bluetooth.
+type bluetoothService struct{}
+
+// Start begins advertising a bluetooth service that acccepts WiFi and Viam cloud config credentials.
+func (bwp *bluetoothService) start(ctx context.Context) error { //nolint:unused
+	// TODO APP-7651: Implement helper methods to start/stop advertising BLE connection
+	bwp.enableAutoAcceptPairRequest() // Async goroutine (hence no error check) which auto-accepts pair requests on this device.
+	return errors.New("TODO APP-7651: Implement helper methods to start/stop advertising BLE connection")
+}
+
+// Stop stops advertising a bluetooth service which (when enabled) accepts WiFi and Viam cloud config credentials.
+func (bwp *bluetoothService) stop() error { //nolint:unused
+	return errors.New("TODO APP-7651: Implement helper methods to start/stop advertising BLE connection")
+}
+
+// Update updates the list of networks that are advertised via bluetooth as available.
+func (bwp *bluetoothService) refreshAvailableNetworks(ctx context.Context, awns []*NetworkInfo) error { //nolint:unused
+	return errors.New("TODO APP-7652: Implement helper method to write update WiFi networks to BLE peripheral characteristic")
+}
+
+// WaitForCredentials returns credentials, the minimum required information to provision a robot and/or its WiFi.
+func (bwp *bluetoothService) waitForCredentials( //nolint:unused
+	ctx context.Context, requiresCloudCredentials, requiresWiFiCredentials bool,
+) (*userInput, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	if !requiresWiFiCredentials && !requiresCloudCredentials {
+		return nil, errors.New("should be waiting for cloud credentials or WiFi credentials, or both")
+	}
+	var ssid, psk, robotPartKeyID, robotPartKey string
+	var ctxErr, ssidErr, pskErr, robotPartKeyIDErr, robotPartKeyErr error
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	utils.ManagedGo(func() {
```

Comment:
I don't understand why you're backgrounding this, only to wg.Wait() right afterwards.